pyproject.toml

```diff
@@ -1,3 +1,21 @@
 [tool.ruff]
-line-length = 88
 target-version = "py312"
+line-length = 88
+
+[tool.ruff.format]
+docstring-code-line-length = 100
+indent-style = "space"  # Use spaces for indentation
+quote-style = "double"
+
+[tool.ruff.lint]
+extend-select = [
+    "E",   # pycodestyle
+    "F",   # Pyflakes
+    "UP",  # pyupgrade
+    "B",   # flake8-bugbear
+    "I",   # isort
```

Comment:
Maybe add RUF100 to the checks? It will make it complain if we tag lint-free lines with `#noqa`

https://docs.astral.sh/ruff/linter/#detecting-unused-suppression-comments